libs/routing/routing_tests/road_access_test.cpp

```diff
@@ -53,77 +52,77 @@ void FillRoadAccessBySample_2(RoadAccess & roadAccess)
-//
-//   RoadAccess::Conditional conditional_4;
-//   conditional_4.Insert(RoadAccess::Type::No, std::move(openingHours[4]));
-//   conditional_4.Insert(RoadAccess::Type::No, std::move(openingHours[5]));
-//
-//   RoadAccess::WayToAccessConditional wayToAccessConditional = {{1 /* featureId */, conditional_1},
-//                                                                {2 /* featureId */, conditional_2}};
-//
-//   RoadAccess::PointToAccessConditional pointToAccessConditional = {
-//       {RoadPoint(3 /* featureId */, 10 /* pointId */), conditional_3},
-//       {RoadPoint(4 /* featureId */, 2 /* pointId */), conditional_4}};
-//
-//   roadAccess.SetAccessConditional(std::move(wayToAccessConditional), std::move(pointToAccessConditional));
-// }
+void FillRoadAccessBySampleConditional_1(RoadAccess & roadAccess)
+{
+  std::vector<std::string> const openingHoursStrings = {"Mo-Su",       "10:00-18:00", "Mo-Fr 10:00-14:00",
+                                                        "09:00-13:00", "Apr - May",   "2010 - 2100"};
+
+  std::vector<osmoh::OpeningHours> openingHours;
+  for (auto const & oh : openingHoursStrings)
+  {
+    openingHours.emplace_back(oh);
+    TEST(openingHours.back().IsValid(), ());
+  }
+
+  RoadAccess::Conditional conditional_1;
+  conditional_1.Insert(RoadAccess::Type::No, std::move(openingHours[0]));
+  conditional_1.Insert(RoadAccess::Type::Private, std::move(openingHours[1]));
+
+  RoadAccess::Conditional conditional_2;
+  conditional_2.Insert(RoadAccess::Type::Destination, std::move(openingHours[2]));
+
+  RoadAccess::Conditional conditional_3;
+  conditional_3.Insert(RoadAccess::Type::No, std::move(openingHours[4]));
+  conditional_3.Insert(RoadAccess::Type::Destination, std::move(openingHours[3]));
+
+  RoadAccess::Conditional conditional_4;
+  conditional_4.Insert(RoadAccess::Type::Destination, std::move(openingHours[5]));
+
+  RoadAccess::WayToAccessConditional wayToAccessConditional = {{1 /* featureId */, conditional_1},
+                                                               {2 /* featureId */, conditional_2}};
+
+  RoadAccess::PointToAccessConditional pointToAccessConditional = {
+      {RoadPoint(3 /* featureId */, 0 /* pointId */), conditional_3},
+      {RoadPoint(4 /* featureId */, 7 /* pointId */), conditional_4}};
+
+  roadAccess.SetAccessConditional(std::move(wayToAccessConditional), std::move(pointToAccessConditional));
+}
+
+void FillRoadAccessBySampleConditional_2(RoadAccess & roadAccess)
+{
+  std::vector<std::string> const openingHoursStrings = {
+      "Mo", "Apr-May 03:00-04:25", "Mo-Sa 12:00-13:00", "2010-2098", "Nov-Apr", "19:00-21:00"};
+
+  std::vector<osmoh::OpeningHours> openingHours;
+  for (auto const & oh : openingHoursStrings)
+  {
+    openingHours.emplace_back(oh);
+    TEST(openingHours.back().IsValid(), (oh));
+  }
+
+  RoadAccess::Conditional conditional_1;
+  conditional_1.Insert(RoadAccess::Type::Private, std::move(openingHours[0]));
```

Comment:
In this way it is clearer which string is converted, no need to guess the array index:
```suggestion
  auto const ToOh = [](std::string_view sv)
  {
    osmoh::OpeningHours oh(sv);
    TEST(oh.IsValid(), (oh));
    return oh;
  };

  RoadAccess::Conditional conditional_1;
  conditional_1.Insert(RoadAccess::Type::Private, ToOh("Mo"));
```